policies/releasestrat.html

```diff
@@ -55,65 +82,92 @@ <h5>
-	    milestone to be closed (see below)</li>
-	    <li>Clean builds in Travis and Appveyor for two days</li>
-	    <li>run-checker.sh to be showing as clean 2 days before release</li>
-	    <li>No open Coverity issues (not flagged as "False Positive" or "Ignore")</li>
-	    <li>TLSv1.3 RFC published (with at least one beta release after the publicaction)</li>
+	    <li>
+	      All open github issues/PRs older than 2 weeks at the time of
+	      release to be assessed for relevance to the version to be
+	      released. Any flagged with the a milestone for the version
+	      to be released must be closed (see below)
+	    </li>
+	    <li>
+	      Clean builds in Travis and Appveyor for two days
+	    </li>
+	    <li>
+	      run-checker.sh to be showing as clean 2 days before release
+	    </li>
+	    <li>
+	      No open Coverity issues (not flagged as "False Positive" or
+	      "Ignore")
+	    </li>
 	  </ul>
 
-	  <p>Valid reasons for closing an issue/PR with a 1.1.1 milestone might be:</p>
+	  <p>
+	    Valid reasons for closing an issue/PR with a milestone for the
+	    version might be:
+	  </p>
+
 	  <ul>
-	    <li>We have just now or sometime in the past fixed the issue</li>
-	    <li>Unable to reproduce (following discussion with original reporter if
-	    possible)</li>
-	    <li>Working as intended</li>
-	    <li>Deliberate decision not to fix until a later release (this wouldn't
-	    actually close the issue/PR but change the milestone instead)</li>
-	    <li>Not enough information and unable to contact reporter</li>
-	    <li>etc</li>
+	    <li>
+	      We have just now or sometime in the past fixed the issue
+	    </li>
+	    <li>
+	      Unable to reproduce (following discussion with original reporter
+	      if possible)
+	    </li>
+	    <li>
+	      Working as intended
+	    </li>
+	    <li>
+	      Deliberate decision not to fix until a later release (this
+	      wouldn't actually close the issue/PR but change the milestone
+	      instead)
+	    </li>
+	    <li>
+	      Not enough information and unable to contact reporter
+	    </li>
 	  </ul>
+
+	  <hr />
+
+	  <p>No API or ABI breaking changes are allowed in a minor or patch release.
+	  The following stability rules apply to all changes made to code
+	  targeted for a major release from version 3.0.0 or later:</p>
+	  <ul>
+	    <li>No existing public interface can be modified except where changes
+	    are unlikely to break source compatibility.
+	    <li>No existing public interface can be removed until its replacement
+	    has been in place in an LTS stable release and the original function has
+	    been documented as deprecated for at least 5 years. A public interface
+	    is any function or macro declared in a public header file.</li>
+	    <li>When structures are made opaque, any newly required accessor
+	    macros or functions are added in a feature release of the extant LTS
+	    release and all supported intermediate successor releases.</li>
```

Comment:
I'm a little confused by this...  does that mean that  will nominate MAJOR releases as LTS and include all MINOR releases therein?

Reply:
An example: let's say we have releases 3.0.x and 3.1.x currently in support and 3.0.x is nominated as LTS. During the development of 3.2.0 we make a structure opaque. This means 3.0.x will have the accessor macros/functions backported to it (because it is an LTS release). They will also be backported to 3.1.x (since it is an intermediate release).

Reply:
At the beginning of  the document, you define MAJOR (no changes within a MAJOR can break API/ABI). But you never defined MINOR.PATCH, though you do say you aren't adopting all of semver, but that is only a statement on what they are not, not what they are. I think they should be defined.

> 3.0.x will have the accessor macros/functions backported to it

This would of course be illegal under semver. Introduction of new features requires a MINOR bump, so only changes that introduce no new features (essentially, bug fixes) can be backported to a MAJOR.MINOR.

What I _think_ you are intending after reading this paragraph and the one on top about MAJOR is this:

MAJOR.STABLE.SERIAL 

- MAJOR: API/ABI incompatible changes bump this
- STABLE: identifies a release line that shall only receive "stable" changes. occaisonally (when?) stable lines will be designated LTS. Both new features and bug fixes are allowed in a stable line (if they are considered stable)
- SERIAL: consecutive releases in a stable line get a new SERIAL number

Is something like this the intention?

You might want to avoid MINOR.PATCH, maybe its my perspective from node/npm which adheres very strongly to semver, but even using those words implies semver to me. But maybe that's not true in other software communities.

Reply:
> An example: let's say we have releases 3.0.x and 3.1.x currently in support and 3.0.x is nominated as LTS. During the development of 3.2.0 we make a structure opaque. This means 3.0.x will have the accessor macros/functions backported to it (because it is an LTS release). They will also be backported to 3.1.x (since it is an intermediate release).

I am confused now. How could you make something opaque in minor release 3.2.0 when this would break API/ABI from 3.0.0?

Reply:
> I am confused now. How could you make something opaque in minor release 3.2.0 when this would break API/ABI from 3.0.0?

Oh. Sorry for the confusion. That's because my example is bad. Let's make the example this instead (I also updated the text above accordingly to save confusing anyone else):

An example: let's say we have releases 3.0.x and 3.1.x currently in support and 3.0.x is nominated as LTS. During the development of 4.0.0 we make a structure opaque. This means 3.0.x will have the accessor macros/functions backported to it (because it is an LTS release). They will also be backported to 3.1.x (since it is an intermediate release).

Reply:
> MAJOR: API/ABI incompatible changes bump this
> STABLE: identifies a release line that shall only receive "stable" changes. occaisonally (when?) stable lines will be designated LTS. Both new features and bug fixes are allowed in a stable line (if they are considered stable)
> SERIAL: consecutive releases in a stable line get a new SERIAL number

Essentially, yes, although I think we will be sticking with MAJOR.MINOR.PATCH since this is already baked into the code. It could of course be changed but it's a more significant change. I think "minor" and "patch" have a meaning outside of semver.

Mostly we only allow bug fixes in a minor release - but there are exceptions. Notably for backport of accessors, and possibly "by exception" for other things.

Reply:
Keeping minor/patch is reasonable, and the overall scheme is, too. I do think text somewhat like the above clearly stating what minor.patch means is missing from the document.

Reply:
I see no reason to go away from MAJOR.MINOR.PATCH.  Those are words for the three positions of a version number that have existed in that form for a looooong time (my memory on this takes me back to the 80's, although as MAJOR.MINOR-PATCH).  Semver is a much later effort to formalize and streamline this form of version number, but that doesn't mean that semver "owns" this version number form or the words used to describe it.

Reply:
I added a description of each of MAJOR, MINOR and PATCH.